flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -155,6 +155,13 @@ private void validate() {
       Preconditions.checkArgument(
           tag == null,
           String.format("Cannot scan table using ref %s configured for streaming reader", tag));
+      Preconditions.checkArgument(
+          snapshotId == null, "Cannot set snapshot-id option for streaming reader");
+      Preconditions.checkArgument(
+          asOfTimestamp == null, "Cannot set as-of-timestamp option for streaming reader");
+      Preconditions.checkArgument(
```

Comment:
where were end snapshot (id or tag) silently overridden? I thought it might be a valid scenario. E.g., during backfill, maybe it can be streaming mode and discover splits snapshot by snapshot with an end snapshot.

Reply:
https://github.com/apache/iceberg/blob/fbcd142c5dc1ec99792ef8edc1378e3a027fecf7/flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/enumerator/ContinuousSplitPlannerImpl.java#L131

Reply:
Here is the code:
```
      Snapshot toSnapshotInclusive =
          toSnapshotInclusive(
              lastConsumedSnapshotId, currentSnapshot, scanContext.maxPlanningSnapshotCount());
      IcebergEnumeratorPosition newPosition =
          IcebergEnumeratorPosition.of(
              toSnapshotInclusive.snapshotId(), toSnapshotInclusive.timestampMillis());
      ScanContext incrementalScan =
          scanContext.copyWithAppendsBetween(
              lastPosition.snapshotId(), toSnapshotInclusive.snapshotId());
```

The `toSnapshotInclusive` reads until the current snapshot. Its only role is to prevent reading more snapshot than `maxPlanningSnapshotCount`. And we set this as a last snapshot for the scan.

Reply:
this is to set the end snapshot per incremental scan/discovery. The source doesn't check/support end snapshot (like Kafka source's [bounded end position](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/kafka/#bounded-ending-position)).

Reply:
Yeah, I got it. This is why we need the new validations to prevent the wrong parametrization of the source.